datastore/storagebackend/postgresql/gettsattrgroups.go

```diff
@@ -0,0 +1,326 @@
+			// TODO: support non-string types, like the 'links' attribute
+			goName := f.Name
+			pbName := common.ToSnakeCase(goName)
+			tspb2go[pbName] = goName
+		}
+	}
+}
+
+// getTSGoNamesFromPBNames returns Go names names corresponding to pbNames.
+func getTSGoNamesFromPBNames(pbNames []string) ([]string, error) {
+	goNames := []string{}
+	for _, pbName := range pbNames {
+		goName, found := tspb2go[pbName]
+		if !found {
+			return nil, fmt.Errorf("pb2go: no value found for key >%s<", pbName)
+		}
+		goNames = append(goNames, goName)
+	}
+	return goNames, nil
+}
+
+// getTSAllPBNames returns names of all database columns.
+func getTSAllPBNames() []string {
+	pbNames := []string{}
+	for pbName := range tspb2go {
+		pbNames = append(pbNames, pbName)
+	}
+	return pbNames
+}
+
+// getTSDBColumns returns names of database columns corresponding to pbNames.
+func getTSDBColumns(pbNames []string) ([]string, error) {
+	seen := map[string]struct{}{}
+	cols := []string{}
+
+	for _, pbName := range pbNames {
+		if _, found := tspb2go[pbName]; !found {
+			return nil, fmt.Errorf(
+				"attribute not found: %s; supported attributes: %s",
+				pbName, strings.Join(getTSAllPBNames(), ", "))
+		}
+
+		if _, found := seen[pbName]; found {
+			return nil, fmt.Errorf("attribute %s specified more than once", pbName)
+		}
+
+		cols = append(cols, pbName)
+		seen[pbName] = struct{}{}
+	}
+
+	return cols, nil
+}
+
+// getTSMdata returns a TSMetadata object initialized from colVals.
+func getTSMdata(colVals map[string]interface{}) (*datastore.TSMetadata, error) {
+
+	tsMData := datastore.TSMetadata{}
+	tp := reflect.ValueOf(&tsMData)
+
+	for pbName, val := range colVals {
+		// NOTE: column names are (assumed to be) identical to protobuf names, hence the
+		// key to the colVals map can be pbName instead of colName
+		goName, found := tspb2go[pbName]
+		if !found {
+			return nil, fmt.Errorf(
+				"key not found in tspb2go: %s (existing contents: %v)", pbName, tspb2go)
+		}
+
+		field := tp.Elem().FieldByName(goName)
+		if !field.IsValid() {
+			return nil, fmt.Errorf("invalid field (goName: %s, pbName: %s)", goName, pbName)
+		}
+		if !field.CanSet() {
+			return nil, fmt.Errorf("unassignable field (goName: %s, pbName: %s)", goName, pbName)
+		}
+
+		switch field.Kind() {
+		case reflect.String:
+			val0, ok := val.(string)
+			if !ok {
+				return nil, fmt.Errorf(
+					"value not string: %v (type: %T; goName: %s, pbName: %s",
+					val, val, goName, pbName)
+			}
+			field.SetString(val0)
+		default:
+			return nil, fmt.Errorf(
+				"unsupported type: %v (val: %v; type: %T; goName: %s; pbName: %s)",
+				field.Kind(), val, val, goName, pbName)
+		}
+	}
+
+	return &tsMData, nil
+}
+
+// scanTsMdata scans column cols from current result row in rows and converts to a
+// TSMetadata object.
+// Returns (TSMetadata object, nil) upon success, otherwise (..., error).
+func scanTsMdata(rows *sql.Rows, cols []string) (*datastore.TSMetadata, error) {
```

Comment:
The function assumes that `rows` contains only the entries in `cols`, correct? So `rows` is the response of a `SELECT` statement for only the columns in `cols`.  Do you want to mention this in the comments?